packages/ses/src/compartment-shim.js

```diff
@@ -51,14 +51,22 @@ export const CompartmentPrototype = {
     } = options;
     const localTransforms = [...transforms];
 
-    const {
-      globalTransforms,
-      globalObject,
-      globalLexicals,
-    } = privateFields.get(this);
+    const compartmentFields = privateFields.get(this);
+    let { globalTransforms } = compartmentFields;
+    const { globalObject, globalLexicals } = compartmentFields;
 
     let localObject = globalLexicals;
     if (__moduleShimLexicals__ !== undefined) {
+      // When using `evaluate` for ESM modules, as should only occur from the
+      // module-shim's module-instance.js, we do not reveal the SES-shim's
+      // module-to-program translation, as this is not standardizable behavior.
+      // However, the `localTransforms` will come from the `__shimTransforms__`
+      // Compartment option in this case, which is a non-standardizable escape
+      // hatch so programs designed specifically for the SES-shim
+      // implementation may opt-in to use the same transforms for `evaluate`
+      // and `import`, at the expense of being tightly coupled to SES-shim.
```

Comment:
Do you have a motivating example in mind?

Reply:
The metering transform.